### PR TITLE
[AMDGPU] Update branch-condition-and.ll to auto-generated checks

### PR DESCRIPTION
**Pull request by jmmartinez**

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/branch-condition-and.ll [hunk 1]
```diff
@@ -1,4 +1,5 @@
-; RUN: llc -mtriple=amdgcn -verify-machineinstrs -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=amdgcn -mcpu=tahiti -verify-machineinstrs -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/branch-condition-and.ll:2]: Don't need -verify-machineinstrs

File: llvm/test/CodeGen/AMDGPU/branch-condition-and.ll [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
 ; RUN: llc -mtriple=amdgcn -mcpu=tonga -mattr=-flat-for-global -verify-machineinstrs -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/branch-condition-and.ll:3]: I'm surprised these got identical output. If you're generating the checks, should lose the -flat-for-global

File: llvm/test/CodeGen/AMDGPU/branch-condition-and.ll [hunk 3]
```diff
@@ -1,5 +1,6 @@
-; RUN: llc -mtriple=amdgcn -verify-machineinstrs -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
-; RUN: llc -mtriple=amdgcn -mcpu=tonga -mattr=-flat-for-global -verify-machineinstrs -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=amdgcn -mcpu=tahiti -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
+; RUN: llc -mtriple=amdgcn -mcpu=tonga -simplifycfg-require-and-preserve-domtree=1 < %s | FileCheck -check-prefix=GCN %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/branch-condition-and.ll:3]: There is now no way these can have identical output. These need different check prefixes

Comment (jmmartinez) [reply llvm/test/CodeGen/AMDGPU/branch-condition-and.ll:3]: If I change the `FileCheck` prefix to `TAHITI` and `TONGA` to see the differences I get:

```
+; TAHITI-LABEL: ham:
+; TAHITI:       ; %bb.0: ; %bb
+; TAHITI-NEXT:    v_cmp_lt_f32_e32 vcc, 0, v0
+; TAHITI-NEXT:    v_cmp_lt_f32_e64 s[0:1], 0, v1
+; TAHITI-NEXT:    s_and_b64 s[0:1], vcc, s[0:1]
+; TAHITI-NEXT:    s_and_saveexec_b64 s[2:3], s[0:1]
+; TAHITI-NEXT:    s_cbranch_execz .LBB0_2
+; TAHITI-NEXT:  ; %bb.1: ; %bb4
+; TAHITI-NEXT:    v_mov_b32_e32 v0, 4
+; TAHITI-NEXT:    s_mov_b32 m0, -1
+; TAHITI-NEXT:    ds_write_b32 v0, v0
+; TAHITI-NEXT:    ; divergent unreachable
+; TAHITI-NEXT:  .LBB0_2: ; %UnifiedReturnBlock
+; TAHITI-NEXT:    s_endpgm
+;
+; TONGA-LABEL: ham:
+; TONGA:       ; %bb.0: ; %bb
+; TONGA-NEXT:    v_cmp_lt_f32_e32 vcc, 0, v0
+; TONGA-NEXT:    v_cmp_lt_f32_e64 s[0:1], 0, v1
+; TONGA-NEXT:    s_and_b64 s[0:1], vcc, s[0:1]
+; TONGA-NEXT:    s_and_saveexec_b64 s[2:3], s[0:1]
+; TONGA-NEXT:    s_cbranch_execz .LBB0_2
+; TONGA-NEXT:  ; %bb.1: ; %bb4
+; TONGA-NEXT:    v_mov_b32_e32 v0, 4
+; TONGA-NEXT:    s_mov_b32 m0, -1
+; TONGA-NEXT:    ds_write_b32 v0, v0
+; TONGA-NEXT:    ; divergent unreachable
+; TONGA-NEXT:  .LBB0_2: ; %UnifiedReturnBlock
+; TONGA-NEXT:    s_endpgm
```

**Review by arsenm — APPROVED** [review]

**Pull request merged**